scripts/backport.py

```diff
@@ -0,0 +1,371 @@
+              }
+            }
+          }
+          """
+        ).substitute({"pr_number": pr_number})
+    )
+
+    # The above returns {'data': {'repository': {'pullRequest': {'closingIssuesReferences': {'edges': [{'node': {'number': 4944}}]}}}}}
+
+    ref_edges = ref_result["data"]["repository"]["pullRequest"][
+        "closingIssuesReferences"
+    ]["edges"]
+
+    if ref_edges:
+        return ref_edges[0]["node"]["number"]
+
+    return None
+
+
+def set_auto_merge(pr_number):
+    """Enable auto-merge for the given PR"""
+
+    owner, name = target_repo_name.split("/")
+    # We first have to find out the PR id, which is some base64 string, different
+    # from its number.
+    query = string.Template(
+        """query {
+          repository(owner: "$owner", name: "$name") {
+            pullRequest(number: $pr_number) {
+              id
+            }
+          }
+        }"""
+    ).substitute(pr_number=pr_number, owner=owner, name=name)
+    result = run_query(query)
+    pr_id = result["data"]["repository"]["pullRequest"]["id"]
+
+    query = string.Template(
+        """mutation {
+            enablePullRequestAutoMerge(
+                input: {
+                    pullRequestId: "$pr_id",
+                    mergeMethod: REBASE
+                }
+            ) {
+                clientMutationId
+            }
+        }"""
+    ).substitute(pr_id=pr_id)
+    run_query(query)
+
+
+# The token has to have the "access public repositories" permission, or else creating a PR returns 404.
+github = Github(os.environ.get("GITHUB_TOKEN"))
+
+# If we are running inside Github Action, will modify the main repo.
+source_remote = "origin"
+source_repo_name = os.environ.get("GITHUB_REPOSITORY")
+target_remote = source_remote
+target_repo_name = source_repo_name
+
+if not source_repo_name:
+    # We are running manually for debugging, probably want to modify a fork.
+    source_repo_name = "timescale/timescaledb"
+    target_repo_name = os.environ.get("BACKPORT_TARGET_REPO")
+    target_remote = os.environ.get("BACKPORT_TARGET_REMOTE")
+    if not target_repo_name or not target_remote:
+        print(
+            "Please specify the target repositories for debugging, using the "
+            "environment variables BACKPORT_TARGET_REPO (e.g. `timescale/timescaledb`) "
+            "and BACKPORT_TARGET_REMOTE (e.g. `origin`).",
+            file=sys.stderr,
+        )
+        sys.exit(1)
+
+print(
+    f"Will look at '{source_repo_name}' (git remote '{source_remote}') for bug fixes, "
+    f"and create the backport PRs in '{target_repo_name}' (git remote '{target_remote}')."
+)
+
+source_repo = github.get_repo(source_repo_name)
+target_repo = github.get_repo(target_repo_name)
+
+# Set git name and email corresponding to the token user.
+token_user = github.get_user()
+os.environ["GIT_COMMITTER_NAME"] = token_user.name
+os.environ[
+    "GIT_COMMITTER_EMAIL"
+] = f"{token_user.id}+{token_user.login}@users.noreply.github.com"
```

Comment:
This is an email that is used by Github when you opt to hide your real email address. It is required so that the commits are recognized by Github as made by the user. That is, if you use a wrong e-mail, there won't be a clickable profile picture next to the commit in the Github interface.